File: pkg/sms/sms.go
package sms

import (
	"bytes"
	"encoding/json"
	"fmt"
	"net/http"

	"github.com/google/uuid"
)

type Sms struct {
	Login    string
	Password string
	Url      string
	Port     string
	Sender   string
}
type SmsRequest struct {
	Messages []SmsReqMessage `json:"messages"`
}

type SmsReqMessage struct {
	Recipient string `json:"recipient"`
	MessageID string `json:"message-id"`
	Sms       struct {
		Originator string `json:"originator"`
		Content    []struct {
			Text string `json:"text"`
		} `json:"content"`
	} `json:"sms"`
}

func NewSmsSender() *Sms {
	return &Sms{
		Login:    "",
		Password: "",
		Url:      "",
		Port:     "",
		Sender:   "",
	}
}

func (s *Sms) SendCode(phone string, text string) error {
	id := uuid.NewString()
	id = "mxb" + string([]rune(id)[:10])
	reqData := SmsRequest{
		Messages: []SmsReqMessage{
			{
				Recipient: phone,
				MessageID: id,
				Sms: struct {
					Originator string `json:"originator"`
					Content    []struct {
						Text string `json:"text"`
					} `json:"content"`
				}{
					Originator: s.Sender,
					Content: []struct {
						Text string `json:"text"`
					}{
						{
							Text: text,
						},
					},
				},
			},
		},
	}
	data, err := json.Marshal(&reqData)
	if err != nil {
		return err
	}
	payloadBuf := new(bytes.Buffer)
	json.NewEncoder(payloadBuf).Encode(data)
	client := &http.Client{}
	req, err := http.NewRequest("POST", s.Url, payloadBuf)
	if err != nil {
		return err
	}
	req.Header.Add("Authorization", fmt.Sprintf("Basic %s:%s", s.Login, s.Password))
	req.Header.Add("Cache-Control", "no-cache")
	req.Header.Add("Content-Type", "application/json")
	res, e := client.Do(req)
	if e != nil {
		return e
	}

	defer res.Body.Close()
	fmt.Println("MESSAGE sended ")
	return nil
}

File: pkg/sms/sms.php
<?php

namespace app\models;


use yii\helpers\Json;

class Sms
{
  private static $login = "mixburger";
  private static $password = "CkcB6I8@z";
  private static $url = "http://91.204.239.44/broker-api/send";
  private static $port = "8083";
  private static $sender = "3700";

  public static function send($phone, $text)
  {
    $id = Dashboard::getGUID();
    $id = "mxb" . substr($id, 0, 10);
    $data = [];
    $data['messages'][] = [
      'recipient' => $phone,
      'message-id' => $id,
      'sms' => [
        'originator' => self::$sender,
        'content' => [
          'text' => $text
        ],
      ],
    ];
    $json = Json::encode($data);
    $curl = curl_init();
    $base = base64_encode(self::$login . ":" . self::$password);

    curl_setopt_array($curl, array(
//            CURLOPT_PORT => self::$port,
      CURLOPT_URL => self::$url,
      CURLOPT_RETURNTRANSFER => true,
      CURLOPT_ENCODING => "",
      CURLOPT_MAXREDIRS => 10,
      CURLOPT_TIMEOUT => 30,
      CURLOPT_HTTP_VERSION => CURL_HTTP_VERSION_1_1,
      CURLOPT_CUSTOMREQUEST => "POST",
      CURLOPT_POSTFIELDS => $json,
      CURLOPT_HTTPHEADER => array(
        "Authorization: Basic {$base}",
        "Cache-Control: no-cache",
        "Content-Type: application/json",
      ),
    ));
    $response = curl_exec($curl);
    $err = curl_error($curl);
    curl_close($curl);
    if ($err) {
      return $err;
    }

    return $response;
  }

}